bio2zarr/vcf2zarr/icf.py

```diff
@@ -458,6 +486,27 @@ def sanitise_value_int_2d(buff, j, value):
         buff[j, :, : value.shape[1]] = value
 
 
+def compute_laa_field(variant) -> list[list[int]]:
+    sample_count = variant.num_called + variant.num_unknown
+    laa_val = [set() for _ in range(sample_count)]
+
+    if "GT" in variant.FORMAT:
+        for sample_index, genotype in enumerate(variant.genotypes):
+            # The last element in the genotype is not an allele.
+            for allele in genotype[:-1]:
+                if allele > 0:
+                    laa_val[sample_index].add(allele)
+    if "AD" in variant.FORMAT:
```

Comment:
What's the reasoning for looking at AD as well? If the alleles aren't referenced in the genotypes, they won't be referenced elsewhere either? I guess you could have alleles with an AD > 0, but are not called in the genotypes, and you therefore want to assign non-zero PL values?

This being the case, we might be better off looking at the non-zero PL values and back-inferring the local alleles - you can't trust implementations to actually follow specifications I'm afraid :(